packages/core/src/billing/billing-address-reducer.ts

```diff
@@ -35,8 +35,9 @@ function dataReducer(
         case BillingAddressActionType.UpdateBillingAddressSucceeded:
         case BillingAddressActionType.ContinueAsGuestSucceeded:
         case CheckoutActionType.LoadCheckoutSucceeded:
-        case OrderActionType.LoadOrderSucceeded:
             return replace(data, action.payload && action.payload.billingAddress);
+        case OrderActionType.LoadOrderSucceeded:
+            return objectMerge(data, action.payload && action.payload.billingAddress);
```

Comment:
This is in depth of the framework/library. Time bomb! ⏲️ 💣

Reply:
Yes, and the funny part was the unit test assume that the mock for getOrder object will have a billing address with valid id.

Reply:
It seems odd that we're replacing or merging order billing address with checkout billing address, as they are essentially different entities, even though their values might be the same. Git history shows that this line was added in #298 to fix a problem. But I think a better fix for that prior problem is to either undo this commit https://github.com/bigcommerce/checkout-sdk-js/pull/283/commits/aba8bd216e82f13006f1889958473df0419e8bf8 or have a separate reducer for order billing address and coupon.

Reply:
I think it's due to us using the same billing address in order-confirmation page as well? In that case shall we have a separate reference to billing-address for order confirmation page? Therefore billing address has a separate reducer for order confirmation page with its own selector?